S/Singular_jll/Compat.toml

```diff
@@ -2,12 +2,22 @@
 JLLWrappers = "1.1.0-1"
 julia = "1"
 
+[401]
+JLLWrappers = "1.1.0-1"
+
 ["401-401.390"]
 FLINT_jll = "200.690.0"
 
 [401-402]
-JLLWrappers = "1.1.0-1"
 julia = "1"
 
-["401.990-402"]
+["401.990-401"]
 FLINT_jll = "200.700.0"
+
+["402.0.0"]
+FLINT_jll = "200.700.0"
+JLLWrappers = "1.1.0-1"
+
+["402.0.1-402"]
+FLINT_jll = "200.700"
```

Comment:
It is my understanding that this corresponds to the compat value `FLINT_jll = "~200.700"` in `Project.toml`. Correct? Then this PR is indeed OK to merge. Thanks @giordano

Reply:
Not 100% sure, but I believe this means "only `200.700.0`".

Reply:
I think the old line above, `FLINT_jll = "200.700.0"`, means "only 200.700.0", i.e./ semver/compat entry `=200.700.00` while this one, which omits the patch part, means `~200.700`, or as a pseudo regex: `200.700.*`.

Otherwise, I don't think it would have rewritten the specifier here compared to lines 18 resp. 15. 

My guess is that the entries here are for use as arguments to a `VersionSpec` constructor; while the compat entries in `Project.toml` are for consumption by `semver_spec` (and until support for `-` ranges was added in Julia 1.4, those compat entries simply could not represent all possible `VersionSpec` values).